src/AggregateFunctions/AggregateFunctionGroupArrayIntersect.cpp

```diff
@@ -0,0 +1,449 @@
+        const auto & offsets = assert_cast<const ColumnArray &>(*columns[0]).getOffsets();
+        const size_t offset = offsets[static_cast<ssize_t>(row_num) - 1];
+        const auto arr_size = offsets[row_num] - offset;
+        ++version;
+        if (version == 1)
+        {
+            for (size_t i = 0; i < arr_size; ++i)
+                map[static_cast<T>((*data_column)[offset + i].get<T>())] = version;
+        }
+        else if (map.size() > 0)
+        {
+            for (size_t i = 0; i < arr_size; ++i)
+            {
+                typename State::Map::LookupResult value = map.find(static_cast<T>((*data_column)[offset + i].get<T>()));
+                if (value != nullptr && value->getMapped() == version - 1)
+                    ++(value->getMapped());
+            }
+        }
+    }
+
+    void merge(AggregateDataPtr __restrict place, ConstAggregateDataPtr rhs, Arena *) const override
+    {
+        auto & map = this->data(place).value;
+        const auto & rhs_map = this->data(rhs).value;
+        UInt64 rhs_version = this->data(rhs).version;
+        if (rhs_version == 0)
+            return;
+        UInt64 version = this->data(place).version++;
+        if (version == 0)
+        {
+            for (auto & rhs_elem : rhs_map)
+            {
+                if (rhs_elem.getMapped() == rhs_version)
+                    map[rhs_elem.getKey()] = 1;
+            }
+            return;
+        }
+        for (auto & rhs_elem : rhs_map)
+        {
+            if (rhs_elem.getMapped() != rhs_version)
+                continue;
+            auto value = map.find(rhs_elem.getKey());
+            if (value != nullptr)
+                ++value->getMapped();
+        }
+    }
+
+    void serialize(ConstAggregateDataPtr __restrict place, WriteBuffer & buf, std::optional<size_t> /* version */) const override
+    {
+        auto & map = this->data(place).value;
+        auto version = this->data(place).version;
+
+        writeVarUInt(version, buf);
+        writeVarUInt(map.size(), buf);
+
+        for (const auto & elem : map)
+        {
+            writeIntBinary(elem.getKey(), buf);
+            writeVarUInt(elem.getMapped(), buf);
+        }
+    }
+
+    void deserialize(AggregateDataPtr __restrict place, ReadBuffer & buf, std::optional<size_t> /* version */, Arena *) const override
+    {
+        readVarUInt(this->data(place).version, buf);
+        this->data(place).value.read(buf);
+    }
+
+    void insertResultInto(AggregateDataPtr __restrict place, IColumn & to, Arena *) const override
+    {
+        ColumnArray & arr_to = assert_cast<ColumnArray &>(to);
+        ColumnArray::Offsets & offsets_to = arr_to.getOffsets();
+
+        const auto & map = this->data(place).value;
+        const auto & version = this->data(place).version;
+        size_t size = 0;
+        for (auto it = map.begin(); it != map.end(); ++it)
+            if (version == it->getMapped())
```

Comment:
Same question as above.
If obsolete versions are stored, the following queries will iterate over the same (obsolete) values repeatedly:
``` sql
SELECT groupArrayIntersect(*) FROM test_numbers;
SELECT groupArrayIntersect(*) FROM test_numbers;
```